defold-rive/commonsrc/tess_renderer.cpp

```diff
@@ -890,82 +387,175 @@ void DefoldTessRenderer::orthographicProjection(float left,
+
+rive::Vec2D* DefoldTessRenderer::getRegionUvs(dmRive::Region* region, float* texcoords_in, int texcoords_in_count)
+{
+    uint32_t tc_index = m_ScratchBufferVec2D.Size();
+    EnsureSize(m_ScratchBufferVec2D, tc_index + texcoords_in_count);
+
+    rive::Vec2D* texcoord_ptr = &m_ScratchBufferVec2D[tc_index];
+
+    dmRive::ConvertRegionToAtlasUV(region, texcoords_in_count/2, texcoords_in, texcoord_ptr);
+    return texcoord_ptr;
+}
+
+void DefoldTessRenderer::drawImageMesh(const rive::RenderImage* _image,
                                       rive::rcp<rive::RenderBuffer> vertices_f32,
                                       rive::rcp<rive::RenderBuffer> uvCoords_f32,
                                       rive::rcp<rive::RenderBuffer> indices_u16,
                                       rive::BlendMode blendMode,
-                                      float opacity) {
-    printf("drawImageMesh\n");
-    // vs_params_t vs_params;
-
-    // const Mat2D& world = transform();
-    // vs_params.mvp = m_Projection * world;
-
-    // setPipeline(m_meshPipeline);
-    // sg_bindings bind = {
-    //     .vertex_buffers[0] = static_cast<SokolBuffer*>(vertices_f32.get())->buffer(),
-    //     .vertex_buffers[1] = static_cast<SokolBuffer*>(uvCoords_f32.get())->buffer(),
-    //     .index_buffer = static_cast<SokolBuffer*>(indices_u16.get())->buffer(),
-    //     .fs_images[SLOT_tex] = static_cast<const SokolRenderImage*>(renderImage)->image(),
-    // };
-
-    // sg_apply_bindings(&bind);
-    // sg_apply_uniforms(SG_SHADERSTAGE_VS, SLOT_vs_params, SG_RANGE_REF(vs_params));
-    // sg_draw(0, indices_u16->count(), 1);
-}
+                                      float opacity)
+{
+    DefoldRenderImage* image = (DefoldRenderImage*)_image;
+
+    dmRive::Region* region = dmRive::FindAtlasRegion(m_Atlas, image->m_NameHash);
+    if (!region)
+    {
+        dmLogError("Couldn't find region '%s' in atlas", dmHashReverseSafe64(image->m_NameHash));
+        return;
+    }
+
+    applyClipping();
+
+    DefoldBuffer<float>*   vertexbuffer = (DefoldBuffer<float>*)vertices_f32.get();
+    DefoldBuffer<uint16_t>* indexbuffer = (DefoldBuffer<uint16_t>*)indices_u16.get();
+
+    DefoldBuffer<float>* uvbuffer = (DefoldBuffer<float>*)uvCoords_f32.get();
+    int count                     = uvbuffer->count();
+    float* uvs                    = uvbuffer->m_Data;
+
+    VsUniforms vs_params = {};
+    vs_params.world = transform();
+
+    FsUniforms fs_uniforms   = {0};
+    fs_uniforms.fillType     = (int) FillType::FILL_TYPE_TEXTURED;
+    fs_uniforms.colors[0][3] = opacity;
+
+    DrawDescriptor desc = {};
+    desc.m_VsUniforms     = vs_params;
+    desc.m_FsUniforms     = fs_uniforms;
+    desc.m_Indices        = indexbuffer->m_Data;
+    desc.m_IndicesCount   = indexbuffer->count();
+    desc.m_Vertices       = (rive::Vec2D*) vertexbuffer->m_Data;
+    desc.m_VerticesCount  = vertexbuffer->count();
+    desc.m_TexCoords      = getRegionUvs(region, uvs, count);
```

Comment:
My biggest beef with this setup is that I think they should handle this internally.
Not much we can do, and this will hopefully be replaced later with their next iteration.